addons/mail/models/mail_message.py

```diff
@@ -1010,13 +1079,14 @@ def read(self, fields=None, load='_classic_read'):
 
     @api.multi
     def write(self, vals):
-        if 'model' in vals or 'res_id' in vals:
+        # todo complete check write invalidation
```

Comment:
it is actually checking that message_type is not a user_notification,we don't need to invalidate record if message_type is a user notification since message_ids as a domain:
domain=lambda self: [('model', '=', self._name), ('message_type', '!=', 'user_notification')], auto_join=True)
So I think it make sense to avoid invalidation when making a message_notify.